### DISCUSSION TITLE
Popuplate list from ViewModel activation

### DISCUSSION THREAD

**Discussion by blemidon** [AvaloniaUI/Avalonia#12615] [Q&A]
I want to populate lists at activation. The Companies property contains the right values, but the combobox on screen remains empty.
My code looks like this:

    public SourceList<CompanyModel> CompaniesSouce { get; private set; }
    private ReadOnlyObservableCollection<CompanyModel> _companies;
    public ReadOnlyObservableCollection<CompanyModel> Companies => _companies;

    public MainViewModel() 
    {
        this.WhenActivated(d =>
        {
            using (MyDbContext context = new())
            {
                List<CompanyModel> qCompanies = context.Companies.ToList();

                CompaniesSouce = new SourceList<CompanyModel>(qCompanies.AsObservableChangeSet());
            }

            CompaniesSouce.Connect()
            .ObserveOn(RxApp.MainThreadScheduler)
            .Bind(out _companies)
            .Subscribe();

            CompaniesSouce.CountChanged.Select(x => x > 0).ToPropertyEx(this, x => x.IsCompanyVisible);

            Disposable
            .Create(() => { })
            .DisposeWith(d);
        });
    }

When I don't use WhenActivated, the program runs fine.

Thanks in advance:
Imre

**Comment by timunie** [comment]
How do you bind to it in XAML?

**Reply by timunie** [reply]
I have to feeling you need to call `RaisePropertyChanged(nameof(Companies))` in order to tell the UI about collection-property has changed

**Comment by blemidon** [comment]
Thank you, I've found the problem. Whenactivated called after UI binding, so the Companies collection was null when the ui binds to it.
I rearrange the code like this, and works:

public MainViewModel() 
{
    CompaniesSouce = new SourceList<CompanyModel>();

        CompaniesSouce.Connect()
        .ObserveOn(RxApp.MainThreadScheduler)
        .Bind(out _companies)
        .Subscribe();

        CompaniesSouce.CountChanged.Select(x => x > 0).ToPropertyEx(this, x => x.IsCompanyVisible);

    this.WhenActivated(d =>
    {
        using (MyDbContext context = new())
        {
            List<CompanyModel> qCompanies = context.Companies.ToList();

            CompaniesSouce.AddRange(qCompanies.AsEnumerable());
        }

        Disposable
        .Create(() => { })
        .DisposeWith(d);
    });
}
